summingbird-online/src/test/scala/com/twitter/summingbird/online/PlannerSpec.scala

```diff
@@ -62,6 +63,30 @@ class PlannerSpec extends WordSpec {
   def arbSource1 = sample[Producer[Memory, Int]]
   def arbSource2 = sample[KeyedProducer[Memory, Int, Int]]
 
+  /*
+ * Test Case : A simple src.map.summer topology with FMMergeableWithSource opt-in functionality.
+ * Asserts : The plan considers the options to convert into a two node toplogy removing FlatMapNode.
+ */
+
+  "The Online Plan with a flat Map which has Summer as dependant and opted-in for FMMergeableWithSource" in {
+    val store1 = testStore
+    val fmname = "flatmapped"
+   // val h1 = arbSource1.map{ i:Int => List(i+1,i+2,i+3) }.flatMap{ x => x.toTraversable }.map{ x => (x *2,1) }.sumByKey(store1)
+    val h1 = arbSource1.flatMap{ i :Int => List((i+1,1),(i+2,1),(i+3,i))}.name(fmname).sumByKey(store1)
+    val opts =  Map(
+      fmname  -> Options().set(FMMergeableWithSource(true))
+    )
+    val planned = Try(OnlinePlan(h1,opts))
+    assert(planned.isSuccess)
```

Comment:
wait, if we assert here and it is false we won't get below to the `isFailure`. I think we need this assert after the if.